src/Core/ExecutionPathTracer/ExecutionPathTracer.cs

```diff
@@ -0,0 +1,307 @@
+        /// </summary>
+        public void OnOperationEndHandler(ICallable operation, IApplyData result)
+        {
+            if (this.nestedTypes.Contains(operation.GetType())) return;
+            this.currDepth--;
+        }
+
+        private static bool IsPartialApplication(ICallable operation)
+        {
+            var t = operation.GetType();
+            if (t == null) return false;
+
+            return t.IsGenericType && t.GetGenericTypeDefinition() == typeof(OperationPartial<,,>);
+        }
+
+        /// <summary>
+        /// Retrieves the <c>QubitRegister</c> associated with the given <c>Qubit</c> or create a new
+        /// one if it doesn't exist.
+        /// </summary>
+        private QubitRegister GetQubitRegister(Qubit qubit)
+        {
+            if (!this.qubitRegisters.ContainsKey(qubit.Id))
+            {
+                this.qubitRegisters[qubit.Id] = new QubitRegister(qubit.Id);
+            }
+
+            return this.qubitRegisters[qubit.Id];
+        }
+
+        private List<QubitRegister> GetQubitRegisters(IEnumerable<Qubit> qubits)
+        {
+            return qubits.Select(this.GetQubitRegister).ToList();
+        }
+
+        /// <summary>
+        /// Creates a new <c>ClassicalRegister</c> and associate it with the given <c>Qubit</c>.
+        /// </summary>
+        private ClassicalRegister CreateClassicalRegister(Qubit measureQubit)
+        {
+            var qId = measureQubit.Id;
+
+            if (!this.classicalRegisters.ContainsKey(qId))
+            {
+                this.classicalRegisters[qId] = new List<ClassicalRegister>();
+            }
+
+            var cId = this.classicalRegisters[qId].Count;
+            ClassicalRegister register = new ClassicalRegister(qId, cId);
+
+            // Add classical register under the given qubit id
+            this.classicalRegisters[qId].Add(register);
+
+            return register;
+        }
+
+        /// <summary>
+        /// Retrieves the most recent <c>ClassicalRegister</c> associated with the given <c>Qubit</c>.
+        /// </summary>
+        /// <remarks>
+        /// Currently not used as this is intended for classically-controlled operations.
+        /// </remarks>
+        private ClassicalRegister GetClassicalRegister(Qubit controlQubit)
+        {
+            var qId = controlQubit.Id;
+            if (!this.classicalRegisters.ContainsKey(qId) || this.classicalRegisters[qId].Count == 0)
+            {
+                throw new Exception("No classical registers found for qubit {qId}.");
+            }
+
+            // Get most recent measurement on given control qubit
+            var cId = this.classicalRegisters[qId].Count - 1;
+            return this.classicalRegisters[qId][cId];
+        }
+
+        /// <summary>
+        /// Given a <c>Type</c> and its value, extract its fields and format it as a string.
+        /// </summary>
+        private string? ExtractArgs(Type t, object value)
```

Comment:
When should this method return null?

Reply:
It returns null if there aren't any arguments found. This was so that the `argStr` value in the resulting `Operation` object would be null and wouldn't be rendered in the final JSON string.

Reply:
That makes sense; I'd suggest including some version of that discussion in the documentation comment, perhaps as a `<returns>` element.